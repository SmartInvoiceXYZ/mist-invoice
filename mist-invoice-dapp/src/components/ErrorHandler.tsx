import { FallbackProps } from "react-error-boundary";

export const ErrorHandler: React.FC<FallbackProps> = ({
  error,
}) => (
  <>
    <h1>Error</h1>
    <pre>{error.message}</pre>
  </>
);
