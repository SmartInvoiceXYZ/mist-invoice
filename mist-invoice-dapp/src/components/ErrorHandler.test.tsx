import { render } from "@testing-library/react";
import React from "react";
import { ErrorHandler } from "./ErrorHandler";

describe("ErrorHandler", () => {
  const mockResetErrorBoundary = jest.fn();

  it("should render", () => {
    const error = new Error("Test error");
    const view = render(
      <ErrorHandler
        error={error}
        resetErrorBoundary={mockResetErrorBoundary}
      />,
    );

    expect(view.asFragment()).toMatchSnapshot();
  });
});
