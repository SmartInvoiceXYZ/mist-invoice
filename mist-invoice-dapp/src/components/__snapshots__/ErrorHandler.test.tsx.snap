// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler should render 1`] = `
<DocumentFragment>
  <h1>
    Error
  </h1>
  <pre>
    Test error
  </pre>
</DocumentFragment>
`;
